### PR TITLE
INGK-570 Improve the tests of the servo module

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

This PR improves the tests of functions shared by all protocols, mainly those from the servo module.

Fixes # INGK-570

### Type of change

- [x] Create a `servo._read_configuration_file` function to read a configuration file. This simplifies the tests of `load_configuration` and `save_configuration`.
- [x] Create a test for the new function.
- [x] Add an XCF reference file to test the `load_configuration` and `save_configuration` functions.
- [x] Improve the tests of `load_configuration` and `save_configuration`.
- [x] Improve the test_write function. 

### Tests
- [x] Run tests and obtain similar coverage.

### CODE REVIEW

File: ingenialink/servo.py [hunk 1]
```diff
@@ -258,6 +245,33 @@ def save_configuration(self, config_file, subnode=None):
         with open(config_file, "wb") as f:
             f.write(dom.toprettyxml(indent='\t').encode())
 
+    def _read_configuration_file(self, config_file):
+        """Read a configuration file. Returns the device metadata and the registers list.
+        Args:
```
Comment (MartinALL91) [inline ingenialink/servo.py:250]: Leave a blank line.

File: ingenialink/servo.py [hunk 2]
```diff
@@ -258,6 +245,33 @@ def save_configuration(self, config_file, subnode=None):
...
+            config_file (str): Path to the dictionary.
+        Returns:
```
Comment (MartinALL91) [inline ingenialink/servo.py:252]: Blank line.

File: ingenialink/servo.py [hunk 3]
```diff
@@ -258,6 +245,33 @@ def save_configuration(self, config_file, subnode=None):
...
+            device: 
+            list: Register list.
+        Raises:
```
Comment (MartinALL91) [inline ingenialink/servo.py:255]: Blank line.

File: ingenialink/servo.py [hunk 4]
```diff
@@ -258,6 +245,33 @@ def save_configuration(self, config_file, subnode=None):
         with open(config_file, "wb") as f:
             f.write(dom.toprettyxml(indent='\t').encode())
 
+    def _read_configuration_file(self, config_file):
```
Comment (MartinALL91) [inline ingenialink/servo.py:248]: This function can be static.

File: tests/test_servo.py [hunk 5]
```diff
@@ -63,14 +57,11 @@ def test_save_configuration(connect_to_slave):
         dtype = saved_register.attrib.get('dtype')
         assert registers[reg_id].dtype == servo.dictionary.dtype_xdf_options[dtype]
 
-        assert access == "rw"
```
Comment (MartinALL91) [inline tests/test_servo.py:66]: I think these assertions should not be removed.

**Review by MartinALL91 — CHANGES_REQUESTED** [review]

**Review by GullinBustin — APPROVED** [review]

**Review by WoodElv — APPROVED** [review]

**Review by MartinALL91 — APPROVED** [review]

**Pull request merged**